File: 0001-two-sum/0001-two-sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var prevMap = [Int: Int]()
        for (i, n) in nums.enumerated() {
            let dif = target - n
            if let index = prevMap[dif] {
                return [index, i]
            }
            prevMap[n] = i
        }
        return []
    }
}
//  prevMap = {} #map that contains both key:value

//         for i,n in enumerate(nums):
//             difference = target - n
//             if difference in prevMap:
//                 return [prevMap[difference],i]
//             prevMap[n]=i
//         return

File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        result = []
        for i in range(len(nums)):
            diff = target - nums[i]
            if diff in nums and nums.index(diff) != i:
                index = nums.index(diff)
                result =[i, index]
                break
            else:
                result=[]
        return result

        